app/models/user.rb

```diff
@@ -274,7 +274,7 @@ def preferred_languages_array
       @preferred_languages_array = language_settings["preferred_languages"].to_a
     else
       languages = []
-      language_settings.keys.each do |setting|
+      language_settings.each_key do |setting|
```

Comment:
`each_key` [doesn't allocate an array](https://github.com/JuanitoFatas/fast-ruby#hasheach_key-instead-of-hashkeyseach-code)